### PR TITLE
【Hackathon 9th No.105】基于Paddle实现CoNFiLD流场生成模型

### PR DESCRIPTION
**Pull request by ADream-ki**
### PR types

Others 

### PR changes

Others 

### Describe

添加CoNFILD案例

### CODE REVIEW

File: docs/zh/examples/confild.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+### 1.3 模型&复现代码
+| 问题类型               | 在线运行                                                                                                                   | 神经网络架构           | 评估指标              |
+|------------------------|----------------------------------------------------------------------------------------------------------------------------|------------------------|-----------------------|
+| 时空湍流生成           | [aistudio](https://aistudio.baidu.com/projectdetail/8933946)                                                                 | 条件神经场+潜在扩散模型 | MSE: 0.041（速度场） |
+
+=== "模型训练命令"
+```bash
+git clone https://github.com/PaddlePaddle/PaddleScience.git
+cd PaddleScience/examples/confild
+python confild.py mode=train
```
Comment (HydrogenSulfate) [inline docs/zh/examples/confild.md]:
```suggestion
python confild.py mode=train
```

File: docs/zh/examples/confild.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+            return data_norm * (
+                params[0].to(data_norm.place) - params[1].to(data_norm.place)
+            ) + params[1].to(data_norm.place)
+        elif method == "ms":
+            return data_norm * params[1].to(data_norm.place) + params[0].to(
+                data_norm.place
+            )
+        elif method == "none":
+            return data_norm
+```
```
Comment (HydrogenSulfate) [inline docs/zh/examples/confild.md]: 文档内的代码片段可以使用引用语法，引用.py文件里的代码

File: docs/zh/examples/confild.md [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+        for i in range(len(self.net1) - 1):
+            x = self.net1[i](x) + self.net2[i](latents)
+            x = self.nl(x)
+        x = self.net1[-1](x)
+        return {self.output_keys[0]: x}
+
+    def disable_gradient(self):
+        for param in self.parameters():
+            param.stop_gradient = not False
```
Comment (HydrogenSulfate) [inline docs/zh/examples/confild.md]: 同上，引用

File: docs/zh/examples/confild.md [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+```
+为了在计算时，准确快速地访问具体变量的值，我们在这里指定网络模型的输入变量名是 ["confild_x", "latent_z"]，输出变量名是 ["confild_output"]，这些命名与后续代码保持一致。
+
+4.3 模型训练、评估
+完成上述设置之后，只需要将上述实例化的对象按照文档进行组合，然后启动训练、评估。
+```python
+def signal_train(cfg, normed_coords, normed_fois, spatio_axis, out_normalizer):
```
Comment (HydrogenSulfate) [inline docs/zh/examples/confild.md]: 同上，引用

File: docs/zh/examples/confild.md [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+                    test_error.append(error)
+                test_error = paddle.concat(x=test_error).mean(axis=0)
+                print("test MAE: ", test_error)
+        if i % 1000 == 0:
+            paddle.save(cnf_model.state_dict(), f"cnf_model_{i}.pdparams")
+            paddle.save(latents_model.state_dict(), f"latents_model_{i}.pdparams")
+```
+
+## 5. 实验结果
+
```
Comment (HydrogenSulfate) [inline docs/zh/examples/confild.md]: 实验结果没有？

File: examples/confild/confild.py
Comment (HydrogenSulfate) [inline examples/confild/confild.py:1]: 注释尽量使用英文

File: examples/confild/confild.py [hunk 6]
```diff
@@ -0,0 +1,656 @@
...
+            return (data_norm + 1) / 2 * (params[0] - params[1]) + params[1]
+        elif method == "01":
+            return data_norm * (params[0] - params[1]) + params[1]
+        elif method == "ms":
+            return data_norm * params[1] + params[0]
+        elif method == "none":
+            return data_norm
+
+
+class basic_set(paddle.io.Dataset):
```
Comment (HydrogenSulfate) [inline examples/confild/confild.py]: 按照一般编程规范，类名建议使用大写`BasicSet`

File: examples/confild/confild.py [hunk 7]
```diff
@@ -0,0 +1,656 @@
...
+    plt.legend()
+
+    # 显示网格线
+    plt.grid(True)
+
+    # 保存为 PNG 格式
+    plt.savefig("case.png")
+
+    # 显示图形
+    plt.show()
```
Comment (HydrogenSulfate) [inline examples/confild/confild.py]: .show可以删除，图片保存即可
```suggestion
```

File: ppsci/arch/confild.py
Comment (HydrogenSulfate) [inline ppsci/arch/confild.py:1]: Image

commit之前请安装pre-commit，并且逐个对上传的文件调用pre-commit run进行格式化

File: ppsci/arch/__init__.py [hunk 8]
```diff
@@ -100,11 +99,15 @@
     "GraphCastNet",
     "HEDeepONets",
     "LorenzEmbedding",
+    "LatentContainer",
     "LatentNO",
     "LatentNO_time",
     "LNO",
+    "LossType",
```
Comment (HydrogenSulfate) [inline ppsci/arch/__init__.py:121]: XXType不需要作为公开接口，可以删除

File: ppsci/arch/confild.py [hunk 9]
```diff
@@ -0,0 +1,1844 @@
...
+        if params is None:
+            params = OrderedDict(self.named_parameters())
+        bias = params.get("bias", None)
+        weight = params["weight"]
+
+        output = paddle.matmul(x=input, y=weight)
+        if bias is not None:
+            output += bias.unsqueeze(axis=-2)
+        return output
+
```
Comment (HydrogenSulfate) [inline ppsci/arch/confild.py:186]: BatchLinear跟nn.Linear的区别是什么，如果没区别是不是可以删掉

File: ppsci/arch/confild.py [hunk 10]
```diff
@@ -0,0 +1,1844 @@
...
+            self.B = pe_init_scale * np.vstack(
+                [(pe_scale**i * np.eye(in_features)) for i in range(pe_num_freqs)]
+            )
+            self.dim = tuple(self.B.shape)[0] * 2
+        elif mode == "rbf":
+            self.centers = paddle.base.framework.EagerParamBase.from_tensor(
+                tensor=paddle.empty(
+                    shape=(rbf_out_features, in_features), dtype="float32"
+                )
+            )
```
Comment (HydrogenSulfate) [inline ppsci/arch/confild.py]: 较新的paddle可以直接用torch类似的语法：self.centers = nn.Paramter(...)

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: docs/zh/examples/confild.md (hunk 1 above)
Comment (ADream-ki) [reply docs/zh/examples/confild.md]: 已修改，请review

File: docs/zh/examples/confild.md (hunk 2 above)
Comment (ADream-ki) [reply docs/zh/examples/confild.md]: 已修改，请review

File: docs/zh/examples/confild.md (hunk 3 above)
Comment (ADream-ki) [reply docs/zh/examples/confild.md]: 已修改，请review

File: docs/zh/examples/confild.md (hunk 4 above)
Comment (ADream-ki) [reply docs/zh/examples/confild.md]: 已修改，请review

File: docs/zh/examples/confild.md (hunk 5 above)
Comment (ADream-ki) [reply docs/zh/examples/confild.md]: 忘了加上去了，请review

File: examples/confild/confild.py (hunk 6 above)
Comment (ADream-ki) [reply examples/confild/confild.py]: 已修改，请review

File: ppsci/arch/__init__.py (hunk 8 above)
Comment (ADream-ki) [reply ppsci/arch/__init__.py:121]: 但是他在外部也是需要使用的，还是说让其作为一个变量传进去？

File: examples/confild/confild.py (hunk 7 above)
Comment (ADream-ki) [reply examples/confild/confild.py]: 已修改，请review

File: ppsci/arch/confild.py (hunk 9 above)
Comment (ADream-ki) [reply ppsci/arch/confild.py:186]: 前者在前向传播时可以动态替换参数

File: ppsci/arch/confild.py (hunk 10 above)
Comment (ADream-ki) [reply ppsci/arch/confild.py]: 已经修改了，请review

**Comment by ADream-ki** [comment]
@HydrogenSulfate 我已经补全了MD文档了，请review
